vpr/src/util/mmap_file.h

```diff
@@ -0,0 +1,18 @@
+#ifndef MMAP_FILE_H_
+#define MMAP_FILE_H_
+
+#include <string>
+#include "capnp/message.h"
+#include "kj/array.h"
+
+class MmapFile {
```

Comment:
Need some significant comments outlining the motivation/use cases for using this vs regular file I/O